From bece701e14926d040c26835051f1d478d84c83a9 Mon Sep 17 00:00:00 2001
From: Ricardo Quesada <ricardoquesada@gmail.com>
Date: Fri, 27 May 2022 06:13:56 -0700
Subject: [PATCH] hci: allow bonding when client doesn't request it

Allow bonding even when the device says that bonding is not required.
This is needed for certain gamepads, like the Nintendo Swich family,
that they support re-connect, but they send an SSP with "bonding not
required"
---
 src/hci.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/hci.c b/src/hci.c
index 6dae414d1..f84b15a8b 100644
--- a/src/hci.c
+++ b/src/hci.c
@@ -3321,6 +3321,7 @@ static void event_handler(uint8_t *packet, uint16_t size){
             if (hci_stack->bondable == false) break;
             // - if security level sufficient
             if (gap_security_level_for_link_key_type(link_key_type) < conn->requested_security_level) break;
+#if 0
             // - for SSP, also check if remote side requested bonding as well
             if (conn->link_key_type != COMBINATION_KEY){
                 bool remote_bonding = conn->io_cap_response_auth_req >= SSP_IO_AUTHREQ_MITM_PROTECTION_NOT_REQUIRED_DEDICATED_BONDING;
@@ -3328,6 +3329,7 @@ static void event_handler(uint8_t *packet, uint16_t size){
                     break;
                 }
             }
+#endif
             gap_store_link_key_for_bd_addr(addr, &packet[8], conn->link_key_type);
             break;
         }
-- 
2.36.1

